tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -908,23 +915,44 @@ pub fn resolve_cuts(
 
                 } else if operation == "neighbors".to_string() {
 
-                    if level_name.level == "Geography".to_string() {
-                        // TODO: Add a better way to identify geography levels in the schema
-                        // TODO: Wait for geoservice API to be ready
-                        return Err(format_err!("Geoservice neighbors not currently supported."));
-                    } else {
-                        let level_cache = match cube_cache.level_caches.get(&level_name.level) {
-                            Some(level_cache) => level_cache,
-                            None => return Err(format_err!("Could not find cached entries for {}.", level_name.level))
-                        };
+                    // Find dimension for the level name
+                    let dimension = match cube.get_dimension(&level_name) {
+                        Some(dimension) => dimension,
+                        None => return Err(format_err!("Could not find dimension for {}.", level_name.level))
+                    };
 
-                        let neighbors_ids = match level_cache.neighbors_map.get(&elements[0]) {
-                            Some(neighbors_ids) => neighbors_ids.clone(),
-                            None => continue
-                        };
+                    match dimension.dim_type {
+                        DimensionType::Geo => {
+                            match geoservice_url {
+                                Some(geoservice_url) => {
+                                    let mut neighbors_ids: Vec<String> = vec![];
+
+                                    let geoservice_response = query_geoservice(geoservice_url, &GeoserviceQuery::Neighbors, &elements[0])?;
```

Comment:
Currently, you're explicitly waiting for the geoservice. This means that a slow geoservice call would block the thread. However, I could see that doing this if/then logic using combinators could be pretty involved, so I'm willing to let it through for now. But this should be a high priority refactor when async/await comes through.

Reply:
If we migrate to actix 1.0, this could also be put inside a `block` section

Reply:
https://github.com/tesseract-olap/tesseract/issues/181